### PR TITLE
Add diagnoser to whylabs-toolkit

### PR DESCRIPTION
**Pull request by christinedraper**
Adds toolkit support for the new diagnoser capability in the WhyLabs API.

Also updates some dependencies for issues found when integrating with the diagnoser capabilities:
* pydantic 1.10.7 caused `TypeError: issubclass() arg 1 must be a class`
* typing-extensions 4.5.0 caused `ImportError: cannot import name 'Doc' from 'typing_extensions'`
* urllib3 2.0.1 caused large (probably compressed) responses to be truncated

Had to adjust test_create_custom_metric to use metric_name not metric_label when deleting. It's an unavoidable consequence of changing to have proper names - we dont want to delete by label.

Bumping minor version rather than patch for the above because of the slightly breaking change and deps as well as significant new feature.

### CODE REVIEW

File: pyproject.toml
Comment (murilommen) [inline pyproject.toml]: is it better to keep it strict to .15?

File: pyproject.toml
Comment (murilommen) [inline pyproject.toml]: this could also go as extra dep for diagnoser only

Comment (christinedraper) [reply pyproject.toml]: It could as long as we dont think the urllib3 2.0.1 problem will impact other uses of the toolkit - I'd worry about large monitor configs, and if its being used in the container. The truncation is really quite a serious issue.

File: pyproject.toml
Comment (christinedraper) [reply pyproject.toml]: unintended... I'll add the "^"

File: pyproject.toml
Comment (murilommen) [reply pyproject.toml]: it used to be a whylogs-container dependency, but I just checked `master` and it's not, so maybe it's not really an issue

File: .bumpversion.cfg
Comment (murilommen) [inline .bumpversion.cfg]: bump one version

File: whylabs_toolkit/monitor/diagnoser/converters/granularity.py
Comment (murilommen) [inline whylabs_toolkit/monitor/diagnoser/converters/granularity.py]: there is a similar "conversion" happening on the helpers package: https://github.com/whylabs/whylabs-toolkit/blob/mainline/whylabs_toolkit/helpers/monitor_helpers.py#L81 maybe we could refactor this to be part of the toolkit in general?

File: whylabs_toolkit/monitor/diagnoser/helpers/utils.py
Comment (murilommen) [inline whylabs_toolkit/monitor/diagnoser/helpers/utils.py:21]: most of this logic is also covered on: https://github.com/whylabs/whylabs-toolkit/blob/mainline/whylabs_toolkit/helpers/config.py

File: whylabs_toolkit/monitor/diagnoser/models/diagnosis_report.py
Comment (murilommen) [inline whylabs_toolkit/monitor/diagnoser/models/diagnosis_report.py:58]: would it be useful to direct extension if any summary/diagnoser class should implement an abstract `describe`?

File: whylabs_toolkit/monitor/diagnoser/recommendation/recommended_change.py
Comment (murilommen) [inline whylabs_toolkit/monitor/diagnoser/recommendation/recommended_change.py:25]: seems that (at least for this class) `self.name` is never reassigned

File: whylabs_toolkit/monitor/diagnoser/targeting.py
Comment (murilommen) [inline whylabs_toolkit/monitor/diagnoser/targeting.py]: why `!= discrete` and not `== continuous`?

File: whylabs_toolkit/monitor/diagnoser/recommendation/recommended_change.py
Comment (christinedraper) [reply whylabs_toolkit/monitor/diagnoser/recommendation/recommended_change.py:25]: this is guarding against being asked to merge in a recommended change of a different type, so it can happen

File: pyproject.toml
Comment (christinedraper) [reply pyproject.toml]: I'm assuming I can leave as is?

File: whylabs_toolkit/monitor/diagnoser/models/diagnosis_report.py
Comment (christinedraper) [reply whylabs_toolkit/monitor/diagnoser/models/diagnosis_report.py:58]: rather hold off on this for now. Not 100% sure describe should be on each item anyway... the result if you compose descriptions hierarchically isnt really what we'd want.

File: whylabs_toolkit/monitor/diagnoser/targeting.py
Comment (christinedraper) [reply whylabs_toolkit/monitor/diagnoser/targeting.py]: that would indeed be better - fixed

File: whylabs_toolkit/monitor/diagnoser/converters/granularity.py
Comment (christinedraper) [reply whylabs_toolkit/monitor/diagnoser/converters/granularity.py]: refactored

File: whylabs_toolkit/monitor/diagnoser/helpers/utils.py
Comment (christinedraper) [reply whylabs_toolkit/monitor/diagnoser/helpers/utils.py:21]: I think the logic in Config is part of why I created env_setup - because it requires one set of env vars; whylogs requires another and I wanted to define them once and pass them in programmatically!  We could promote env setup into helpers if you think that makes sense

**Review by murilommen — APPROVED** [review]
looks good :) 
PS: remember to use dev0 only on this branch to do dev releases and test it further and then `make bump-release` before merging it to main

**Pull request merged**
